pyramid_zipkin/tween.py

```diff
@@ -192,23 +192,33 @@ def tween(request: Request) -> Response:
             tween_kwargs['firehose_handler'] = zipkin_settings.firehose_handler
 
         with tracer.zipkin_span(**tween_kwargs) as zipkin_context:
-            response = handler(request)
-            if zipkin_settings.use_pattern_as_span_name and request.matched_route:
-                zipkin_context.override_span_name('{} {}'.format(
-                    request.method,
-                    request.matched_route.pattern,
-                ))
-            zipkin_context.update_binary_annotations(
-                get_binary_annotations(request, response),
-            )
-
-            if zipkin_settings.post_handler_hook:
-                zipkin_settings.post_handler_hook(
-                    request,
-                    response,
-                    zipkin_context
+            response = None
+            try:
+                response = handler(request)
+            except Exception as e:
+                zipkin_context.update_binary_annotations({
+                    'error_type': type(e).__name__,
+                    'response_status_code': '500'
```

Comment:
Do we assume that all the exceptions are 500? I think we may have different types of exceptions that can be mapped with other status code like 404 or 400.
 I'm not sure if it's worth the effort to implement the mapping logic?
 This is kind of good enough for me.

Reply:
>Do we assume that all the exceptions are 500?
This PR set response code largely for the unhandled exceptions (and any other exceptions thrown explicitly by applications that are of type Exception if these do exist). And the reason 500 seems justified is if we dont handle it here the outer level uwsgi exception handler will do so and throw an HTTPInternalServerError (which is a 500)
https://sourcegraph.yelpcorp.com/python-packages/yelp_pyramid/-/blob/yelp_pyramid/uwsgi_exception_handler.py?L39

 >I think we may have different types of exceptions that can be mapped with other status code like 404 or 400.
 We should just have to worry about SERVER side error codes here?
https://docs.pylonsproject.org/projects/pyramid/en/latest/api/httpexceptions.html
Based on this list ^ most of these exceptions are not something we would be expecting application code to raise.